features/dd-sdk-android-webview/src/test/kotlin/com/datadog/android/webview/internal/replay/WebViewReplayEventConsumerTest.kt

```diff
@@ -0,0 +1,288 @@
+import com.google.gson.JsonObject
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.annotation.LongForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class WebViewReplayEventConsumerTest {
+
+    private lateinit var testedConsumer: WebViewReplayEventConsumer
+
+    @Forgery
+    lateinit var fakeMappedEvent: JsonObject
+
+    @LongForgery
+    var fakeServerTimeOffsetInMillis: Long = 0L
+
+    @Mock
+    lateinit var mockWebViewReplayMapper: WebViewReplayEventMapper
+
+    @Mock
+    lateinit var mockRumContextProvider: WebViewRumEventContextProvider
+
+    @Mock
+    lateinit var mockSdkCore: InternalSdkCore
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Mock
+    lateinit var mockSessionReplayFeatureScope: FeatureScope
+
+    @Forgery
+    lateinit var fakeDatadogContext: DatadogContext
+
+    @Forgery
+    lateinit var fakeRumContext: RumContext
+
+    lateinit var fakeSessionReplayFeatureContext: Map<String, Any?>
+
+    lateinit var fakeValidBrowserEvent: JsonObject
+
+    @Mock
+    lateinit var mockDataWriter: DataWriter<JsonObject>
+
+    @Mock
+    lateinit var mockEventBatchWriter: EventBatchWriter
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeSessionReplayFeatureContext = forge.aMap {
+            WebViewReplayEventConsumer.SESSION_REPLAY_ENABLED_KEY to true
+        }
+        fakeValidBrowserEvent = forge.getForgery()
+        fakeRumContext = fakeRumContext.copy(
+            sessionState =
+            WebViewReplayEventConsumer.SESSION_TRACKED_STATE
+        )
+        fakeDatadogContext = fakeDatadogContext.copy(
+            time = fakeDatadogContext.time.copy(
+                serverTimeOffsetMs = fakeServerTimeOffsetInMillis
+            ),
+            featuresContext = forge.aMap {
+                Feature.SESSION_REPLAY_FEATURE_NAME to fakeSessionReplayFeatureContext
+            }
+        )
+        whenever(
+            mockSdkCore.getDatadogContext()
+        ) doReturn fakeDatadogContext
```

Comment:
this shouldn't be needed, because you get `DatadogContext` through the `withWriteContext` call.